### ISSUE TITLE
dart:html should not be used anymore. Switch to package:web?

### ISSUE THREAD

**Issue by eggnstone** [poirierlouis/file_system_access_api#9]
When using `file_system_access_api` e.g. `html.window.showOpenFilePicker()` a warning is given:

`avoid_web_libraries_in_flutter`
https://dart.dev/tools/diagnostic-messages#avoid_web_libraries_in_flutter
Don't use web-only libraries outside Flutter web plugin packages.

Will you switch the package to use `package:web`?

**Comment by poirierlouis** [comment]
I'll take a look, thanks for reporting.

**Comment by poirierlouis** [comment]
@ditman, regarding conversation on [path_provider](https://github.com/flutter/flutter/issues/45296#issuecomment-2899828647), I'd appreciate your advice:

> (PS: file_system_access_api wouldn't be usable for flutter/packages as is right now, because it's implemented with dart:html and it won't compile to WASM, however package:web offers pretty much direct access to the API documented in MDN, so it's not the end of the world)

I haven't updated the package in deed. I wonder however if `file_system_access_api` is now obsolete with the introduction of `package:web`? Though iirc, `package:web` only implement standardized features, meaning it doesn't include all features exposed by `file_system_access_api`, right?

**Comment by ditman** [comment]
> Though iirc, package:web only implement standardized features, meaning it doesn't include all features exposed by file_system_access_api, right?

This is correct, `package:web` is going to give you the js-interop API with the standard JS features from the browser (what you can find on MDN, for example). All the extra features you added, like checking if the feature is supported, you need to continue providing, so for example:

See this:
* https://dart.dev/interop/js-interop/package-web

The only "complicated" feature is that there's now a separation between JS types and Dart types (`JSString` vs `String`), and you need to be aware of what is flowing in your code. In my experience, I would *not* expose any JS-interop types directly to end users of the package if I can avoid it, and instead create the necessary objects in pure Dart. Then have an internal way of converting between the two. We have a lot of this in `package:google_maps_flutter_web`, if you want to take a look!

**Comment by poirierlouis** [comment]
Alright thank you for the explanation and the package to look at. I see, so yeah this package can still be relevant to only provide pure Dart types to users (in addition to extra features).
I'll be able to refactor it then.

**Comment by mrcsh** [comment]
@poirierlouis , I have done an initial port to packag:web/js_interop.  It is not complete but has what I need for my app.  
Let me know if you want me to submit a PR or just point you at the repo if you want to use it as a basis for a more complete effort.

**Comment by poirierlouis** [comment]
@mrcsh, thank you!
I took a quick look for now and it looks promising. If I understood, your port use a Dart wrapper for JS handles? Is this required due to the new `package:web`/`package:js_interop`?

**Comment by mrcsh** [comment]
I don't think it's strictly necessary but without it everyone using the JS objects directly would have to be deal with all the .toDart and .toJS conversions which is tedious at best.   I'm not a web developer and only know enough JS/HTML to be dangerous and it took some trial and error to get things right.
BTW how does one run the example code? It does not appear to be a standard flutter app.

**Comment by poirierlouis** [comment]
Hum, my first implementation was using wrappers indeed, until I learn how it worked (in past SDK). As long as the API is backward compatible for DX, that's fine.
It's not indeed, it is a plain Dart/web implementation. You should be able to run a light HTTP server based on the README in example/.
